openmc/mgxs/mdgxs.py

```diff
@@ -0,0 +1,1564 @@
+
+        # Slice energy groups if needed
+        filters = []
+        filter_bins = []
+
+        if len(groups) != 0:
+            energy_bins = []
+            for group in groups:
+                group_bounds = self.energy_groups.get_group_bounds(group)
+                energy_bins.append(group_bounds)
+            filter_bins.append(tuple(energy_bins))
+            filters.append('energyout')
+
+        if len(delayed_groups) != 0:
+            filter_bins.append(tuple(delayed_groups))
+            filters.append('delayedgroup')
+
+        if filters != []:
+
+            # Slice nu-fission-out tally along energyout filter
+            delayed_nu_fission_out = slice_xs.tallies['delayed-nu-fission-out']
+            tally_slice = delayed_nu_fission_out.get_slice\
+                          (filters=filters, filter_bins=filter_bins)
+            slice_xs._tallies['delayed-nu-fission-out'] = tally_slice
+
+        # Add energy filter back to nu-fission-in tallies
+        self.tallies['delayed-nu-fission-in'].add_filter(energy_filter)
+        slice_xs._tallies['delayed-nu-fission-in'].add_filter(energy_filter)
+
+        slice_xs.sparse = self.sparse
+        return slice_xs
+
+    def merge(self, other):
+        """Merge another ChiDelayed with this one
+
+        If results have been loaded from a statepoint, then ChiDelayed are only
+        mergeable along one and only one of energy groups or nuclides.
+
+        Parameters
+        ----------
+        other : openmc.mdgxs.MDGXS
+            MDGXS to merge with this one
+
+        Returns
+        -------
+        merged_mdgxs : openmc.mgxs.MDGXS
+            Merged MDGXS
+        """
+
+        if not self.can_merge(other):
+            raise ValueError('Unable to merge ChiDelayed')
+
+        # Create deep copy of tally to return as merged tally
+        merged_mdgxs = copy.deepcopy(self)
+        merged_mdgxs._derived = True
+        merged_mdgxs._rxn_rate_tally = None
+        merged_mdgxs._xs_tally = None
+
+        # Merge energy groups
+        if self.energy_groups != other.energy_groups:
+            merged_groups = self.energy_groups.merge(other.energy_groups)
+            merged_mdgxs.energy_groups = merged_groups
+
+        # Merge delayed groups
+        if self.delayed_groups != other.delayed_groups:
+            merged_delayed_groups = self.delayed_groups.merge\
+                                    (other.delayed_groups)
+            merged_mdgxs.delayed_groups = merged_delayed_groups
+
+        # Merge nuclides
+        if self.nuclides != other.nuclides:
+
+            # The nuclides must be mutually exclusive
+            for nuclide in self.nuclides:
+                if nuclide in other.nuclides:
+                    msg = 'Unable to merge Chi Delayed with shared nuclides'
+                    raise ValueError(msg)
+
+            # Concatenate lists of nuclides for the merged MGXS
+            merged_mdgxs.nuclides = self.nuclides + other.nuclides
```

Comment:
Hmmm, is this really how we have been merging nuclides? Seems like we should eliminate any common nuclides.